services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchSpec.groovy

```diff
@@ -24,4 +33,118 @@ class SwitchSpec extends BaseSpecification {
         def exc = thrown(HttpClientErrorException)
         exc.rawStatusCode == 400
     }
+
+    def "Delete nonexistent switch"() {
+        given: "Nonexistent switch ID"
+        def switchId = new SwitchId("de:ad:be:ef:99:99:99:99")
+
+        when: "Try to delete nonexistent switch"
+        northboundService.deleteSwitch(switchId, false)
+
+        then: "Got 404 NotFound"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 404
+    }
+
+    def "Delete active switch"() {
+        given: "An active switch"
+        def sw = topology.getActiveSwitches()[0]
+
+        when: "Try to delete active switch"
+        northboundService.deleteSwitch(sw.getDpId(), false)
+
+        then: "Got 400 BadRequest because switch must be deactivated"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+    }
+
+    def "Delete inactive switch with ISLs"() {
+        requireProfiles("virtual")
+
+        given: "An active switch"
+        def sw = topology.getActiveSwitches()[0]
+
+        and: "Deactivate this switch"
```

Comment:
Currently this better fits for a 'when' block description, not 'given'. 
Should probably be sth like
```
given: "Deactivated switch"
when: "Try to delete inactive switch with ISLs"
```
or
```
given: "An active switch"
when: "Deactivate this switch"
and: "Try to delete inactive switch with ISLs"
```